### DISCUSSION TITLE
I didn't understand

### DISCUSSION THREAD

**Discussion by OurCodeBase** [AmrDeveloper/CodeView#36] [General]
### Dear @AmrDeveloper Sir,
Sorry to disturbing you but i really want help!
`codeView.addSyntaxPattern(pattern, Color);`
Can you please tell me what is `pattern` and `Color` in this code ? And how to use them ?

**Comment by AmrDeveloper** [comment]
Hello @harsh7i, 

The idea behind CodeView is that you add a set of patterns and colors, and CodeView collect them into a `Map<Pattern, Color>` once the user write any text it will be highlighted depending on the color of the pattern that matches this text, for example if you added a pattern for number with color red, and pattern for string literal with color blue, once you write any text that matches the pattern of number for example 1 or 10 it will be red, 

Please check this package [Syntax Package](https://github.com/AmrDeveloper/CodeView/tree/master/app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax) in the example app you will find a set of patterns for popular programming languages

If you have any question please ask it and we can talk about it
